angelsindustries/control.lua

```diff
@@ -65,179 +66,373 @@ local function initialize_crash_site()
+      local toggled = force.ghost_time_to_live == 0
+      force.ghost_time_to_live = toggled and 60 * 60 * 60 * 24 * 7 or 0
 
-    ghosting[research.name] = nil
-    for tech in pairs(ghosting) do
-      if research.force.technologies[tech] then
-        if not research.force.technologies[tech].researched then
-          research.force.technologies[tech].researched = true
-          research.force.technologies[tech].enabled = false
-        end
+      for _, fplayer in pairs(force.players) do
+        fplayer.set_shortcut_toggled(input, toggled)
       end
     end
+  end
+)
 
-  else -- not a ghosting tech (but could be the final prerequisite in order to be researchable)
-    -- this makes sure only 1 will be available for research, the others will be disabled
-    local already_available = false
-    for tech,_ in pairs(ghosting) do
-      if research.force.technologies[tech] and (not research.force.technologies[tech].researched) then
+-- Vehicle burner equipment
 
-        local available = true
-        for _,prereq in pairs(research.force.technologies[tech].prerequisites or {}) do
-          if not prereq.researched then
-            available = false
-          end
-        end
+script.on_init(
+  function(event)
+    global.vehicle_burners = {}
+    global.current_car = {}
+  end
+)
+
+script.on_configuration_changed(
+  function(event)
+    if global.vehicle_burners == nil then
+      global.vehicle_burners = {}
+    end
+    if global.current_car == nil then
+      global.current_car = {}
+    end
+  end
+)
+
+script.on_event(
+  defines.events.on_gui_opened,
+  function(event)
+    if
+      event.gui_type == defines.gui_type.entity and event.entity ~= nil and
+        event.entity.type == "car"
+     then
+      global.current_car = event.entity
+    end
+  end
+)
 
-        if available then -- shows up as researchable
-          if already_available then
-            --research.force.technologies[tech].researched = true
-            research.force.technologies[tech].enabled = false
+script.on_event(
+  defines.events.on_gui_closed,
+  function(event)
+    if
+      event.gui_type == defines.gui_type.entity and event.entity.type == "car" and
+        global.current_car == event.entity
+     then
+      global.current_car = {}
+    end
+  end
+)
+
+script.on_event(
+  defines.events.on_player_placed_equipment,
+  function(event)
+    if event.equipment.name == "angels-burner-generator-vequip" and next(global.current_car) ~= nil then
+      global.vehicle_burners[global.current_car.unit_number] = {
+        vehicle = global.current_car,
+        burner = event.equipment,
+        grid = event.grid
+      }
+    end
+  end
+)
+
+script.on_event(
+  defines.events.on_player_removed_equipment,
+  function(event)
+    if event.equipment == "angels-burner-generator-vequip" and next(global.current_car) ~= nil then
+      global.vehicle_burners[global.current_car.unit_number] = nil
+    end
+  end
+)
+
+script.on_nth_tick(
+  60,
```

Comment:
I'm not sure if we want to increase the tick count a bit higher but im fine with it now.
The thing to consider, is realistically, how many "stacks" of wood do we want to stick in per second?